### PR TITLE
644 7e5 create workspace service

### PR DESCRIPTION
**Pull request by AdrianaGroza**

### CODE REVIEW

File: src/services/workspace-service.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		method: "GET",
+		headers: {
+			Authorization: `Bearer ${token}}`,
+		},
+	});
+		
+    if (response.status === 200) {
+        const data = await response.json();
+        localStorage.setWorkSpace("token", data.token);
+        dispatch(setWorkspace(data));
```
Comment (cezarmocanu) [inline src/services/workspace-service.js]:
```js
return data; 
```

Comment (cezarmocanu) [inline src/services/workspace-service.js]: dispatch is undefined

File: src/services/workspace-service.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      } else {
+        dispatch(setWorkspace([]));
```
Comment (cezarmocanu) [inline src/services/workspace-service.js]: dispatch is undefined
setWorkspace can only be used inside a react component

File: src/services/workspace-service.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { ENDPOINTS } from "../constants/api";
+import { setWorkspace } from "../store/slices/workspace-slice";
+
+const getWorkspace = async () => {
```
Comment (cezarmocanu) [inline src/services/workspace-service.js:4]: rename into getWorkspaces because it is a list

File: src/store/slices/workspace-slice.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const workspaceSlice = createSlice({
+    name: 'workspace',
+    initialState: [],
+    reducers: {
+        setWorkspace: (state, action) => {
+        return action.payload;
```
Comment (cezarmocanu) [inline src/store/slices/workspace-slice.js:8]: state = action.payload;
return state;

File: src/constants/api.js [hunk 5]
```diff
@@ -4,4 +4,7 @@ export const ENDPOINTS = {
 	AUTH: {
```
Comment (cezarmocanu) [inline src/constants/api.js:4]: You can add the code below insede the test component to check if your service works properly
```js
useEffect(() => {
		workspaceService.getWorkspace();
	}, []);
```

File: src/services/workspace-service.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+	const token = localStorage.getItem("token");
+
+	const response = await fetch(ENDPOINTS.AUTH.WORKSPACE.USER_WORKSPACE, {
```
Comment (cezarmocanu) [inline src/services/workspace-service.js]: remove ```AUTH``` form object

File: src/services/workspace-service.js (hunk 2 above)
Comment (cezarmocanu) [reply src/services/workspace-service.js]: return [] in case of error or other status

File: src/constants/api.js (hunk 5 above)
Comment (cezarmocanu) [reply src/constants/api.js:4]: Also modiffy this useEffect hook to dispaly the data received from the service function

**Review by cezarmocanu — CHANGES_REQUESTED** [review]

File: src/store/slices/workspace-slice.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+        }
+    }
+  });
+
+export const { setWorkspace } = workspaceSlice.actions;
+
```
Comment (AleBarboni) [inline src/store/slices/workspace-slice.js:14]: remove white space

**Review by cezarmocanu — APPROVED** [review]

**Review by HalaszPaul — APPROVED** [review]

**Review by AleBarboni — APPROVED** [review]

**Review by ramona5lodea — APPROVED** [review]

**Pull request merged**